src/ray/object_manager/test/object_manager_test.cc

```diff
@@ -236,6 +238,82 @@ class TestObjectManagerCommands : public TestObjectManager {
 
   void NotificationTestComplete(ObjectID object_id_1, ObjectID object_id_2) {
     ASSERT_EQ(object_id_1, object_id_2);
+    NextWaitTest();
+  }
+
+  void NextWaitTest() {
+    current_wait_test += 1;
+    switch(current_wait_test){
+      case 0: {
+        TestWait(100, 5, 3, 0, false, false);
+      } break;
+      case 1: {
+        TestWait(100, 5, 3, 1000, false, true);
+      } break;
+      case 2: {
+        TestWait(100, 5, 3, 1000, false, false);
+      } break;
+      case 3: {
+        TestWait(100, 5, 6, 1000, true, false);
+      } break;
+    }
+  }
+
+  void TestWait(int data_size, int num_objects, uint64_t required_objects, int wait_ms, bool include_nonexistent, bool test_local) {
+    std::vector<ObjectID> object_ids;
+    for (int i = -1; ++i < num_objects;) {
+      ObjectID oid;
+      if (test_local) {
+        oid = WriteDataToClient(client1, data_size);
+      } else {
+        oid = WriteDataToClient(client2, data_size);
+      }
+      object_ids.push_back(oid);
+    }
+    if (include_nonexistent) {
+      num_objects += 1;
+      object_ids.push_back(ObjectID::from_random());
+    }
+    server1->object_manager_.Wait(
+        object_ids, wait_ms, required_objects, false,
+        [this, num_objects, wait_ms, required_objects](int64_t elapsed,
+                            const std::unordered_set<ray::ObjectID> &found,
+                            const std::unordered_set<ray::ObjectID> &remaining) {
+          switch(current_wait_test){
+            case 0: {
+              ASSERT_TRUE(elapsed == 0);
```

Comment:
Checking for `elapsed == 0` seems like it will be pretty flaky when run on a CI. Also, if we can, we should avoid exposing things that are only used for testing purposes as part of the API (e.g., exposing `elapsed` as an argument for the wait callback).

Reply:
The implementation explicitly sets elapsed to zero when `wait_ms` is 0, so this should never flake. Will look into whether it makes sense to remove elapsed.

Reply:
Okay, but I still think we should remove `elapsed`. We should write interfaces that we can unit test, but we should not write interfaces that are only for unit testing.